app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
+  end
+
+  def loa_no_js_fallback_link
+    if loa1_signup_context?
+      destroy_user_path
+    elsif loa3_context?
+      verify_session_path
+    end
```

Comment:
We are eventually going to want everyone to be able to cancel sign ups. I will just change the method names and add whatever other logic needed to detect these cases.